src/ParameterDistributions.jl

```diff
@@ -0,0 +1,321 @@
+    #Distinguish 1 sample of an ND parameter or N samples of 1D parameter, and store as 2D array  
+    Samples(distribution_samples::Array{FT,1}; params_are_columns=true) where {FT <: Real} = params_are_columns ? new{FT}(reshape(distribution_samples,1,:)) : new{FT}(reshape(distribution_samples,:,1))
+    
+end 
+
+
+# For the transforms
+abstract type ConstraintType end
+
+"""
+    NoConstraint <: ConstraintType
+
+No constraint.
+"""
+
+struct NoConstraint <: ConstraintType end
+
+"""
+    BoundedBelow{FT <: Real} <: ConstraintType
+
+A lower bound constraint.
+"""
+struct BoundedBelow{FT <: Real} <: ConstraintType
+    lower_bound::FT
+end
+"""
+    BoundedAbove{FT <: Real} <: ConstraintType
+
+And upper bound constraint
+"""
+struct BoundedAbove{FT <: Real} <: ConstraintType
+    upper_bound::FT
+end
+
+"""
+    Bounded{FT <: Real} <: ConstraintType
+
+Both a lower and upper bound constraint.
+"""
+struct Bounded{FT <: Real} <: ConstraintType
+    lower_bound::FT
+    upper_bound::FT
+    Bounded(lower_bound::FT,upper_bound::FT) where {FT <: Real} = upper_bound <= lower_bound ? throw(DomainError("upper bound must be greater than lower bound")) : new{FT}(lower_bound, upper_bound)
+end
+
+"""
+    len(c::Array{CType})
+
+The number of constraints, each constraint has length 1.
+"""
+function len(c::CType) where {CType <: ConstraintType}
+    return 1
+end
+
+function len(carray::Array{CType}) where {CType <: ConstraintType}
+    return size(carray)[1]
+end
+
+"""
+    dimension(d::ParametrizedDistributionType)
+
+The number of dimensions of the parameter space
+"""
+function dimension(d::Parameterized) 
+    return length(d.distribution)
+end
+
+function dimension(d::Samples)
+    return size(d.distribution_samples)[1]
+end
+
+"""
+    n_samples(d::Samples)
+
+The number of samples in the array
+"""
+function n_samples(d::Samples)
+    return size(d.distribution_samples)[2]
+end
+
+"""
+    struct ParameterDistribution
+
+Structure to hold a parameter distribution
+"""
+struct ParameterDistribution{PDType <: ParameterDistributionType, CType <: ConstraintType}
+    distribution::PDType
+    constraints::Array{CType}
+    name::String
+
+    ParameterDistribution(parameter_distribution::PDType,
+                          constraints::Array{CType},
+                          name::String) where {PDType <: ParameterDistributionType,
+                                               CType <: ConstraintType} = !(dimension(parameter_distribution) ==
+len(constraints)) ? throw(DimensionMismatch("There must be one constraint per parameter")) : new{PDType,CType}(parameter_distribution, constraints, name)     
+end
+    
+
+"""
+    struct ParameterDistributions
+
+Structure to hold an array of ParameterDistribution's
+"""
+struct ParameterDistributions
+    parameter_distributions::Array{ParameterDistribution}
+end
+
```

Comment:
What practical use do you envision for this structure? What benefits does it bring compared to for example, creating another `ParameterDistribution` constructor that concatenates parameters/distributions in the same object?

Reply:
I guess i thought maybe the modularity might be useful, maybe if the distributions are all stored as different `Parameterized` and `Samples` or KDE type objects. For examples you could have a parameter distribution for each different part of Clima and then when we run CES we just combine them with 

`ParameterDistributions(Convection_dist, Microphysics_dist, EDMF_dist, Ocean_surface_dist)`

But I think your suggestion also does this. Do you have a preference?

Reply:
As discussed, let's try to use another constructor of `ParameterDistribution` for now.

Reply:
Done